coalesce.go

```diff
@@ -29,3 +29,12 @@ func CoalesceT(values ...time.Time) (v time.Time, ok bool) {
 func MustCoalesceT(values ...time.Time) time.Time {
 	return MustFirstT(NonZeroT, values...)
 }
+
+// Coalescer returns a DurationTransform that returns the first value positive
+// value. The transform input value is checked first, then each of the given
+// values.
```

Comment:
```suggestion
// Coalescer returns a DurationTransform that returns the first positive value.
// The transform input value is checked first, then each of the given values.
```

You may have also meant "that returns the first value that is a positive value" ?

Reply:
This first sentence is still kind of awkward?
> returns a DurationTransform that returns

Reply:
I guess it returns an object/function thing that when called will then return another thing. Maybe the sentence is fine after that "value positive value" suggestion is applied.

Reply:
So you were happy with the original doc block?

Reply:
I could reword it like so, if you think it's better:

```go
// Coalescer returns a DurationTransform that selects the first positive value
// from the transform input value and the given values.
//
// It panics if none of the values are positive.
func Coalescer(values ...time.Duration) DurationTransform {
```

Reply:
Or perhaps:

```go
// Coalescer returns a DurationTransform that performs a Coalesce() on the
// transform input value with the given values.
//
// It panics if neither the input value nor any of the given values are positive.
```

As an aside, there are two other functions introduced in this PR that currently use the `returns a DurationTransform that returns` language.

Reply:
Ok, I pushed some changes (different to what I suggested above), see what you think.

Reply:
I don't think you actually pushed any changes?

The main issue I found and did the suggestion fix for was the "_value_ ... _value_" usage. Is that fix not correct for your intended description?
> returns the first value positive value.

The "_returns a BlahTransform that returns ..._" wasn't so bad once i realised that the Transform thing was someting else that gets used to return another thing.

Reply:
Ah, you're right - I did it on the wrong branch >_<.

Reply:
No wonder you were surprised that I was happy with those docs :P